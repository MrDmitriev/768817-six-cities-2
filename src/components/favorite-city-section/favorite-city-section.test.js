import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';

import {FavoriteCitySection} from './favorite-city-section.jsx';
import history from '../../history/history.js';

it(`should match snapShot`, () => {
  const props = {
    city: `AAA`,
    onBookmarkClick: jest.fn(),
    offers: [{
      name: `AAA`,
      price: 120,
      type: `AAA`,
      images: [`aaa`],
    }]
  };

  const wrapper = renderer.create(
      <Router history={history}>
        <FavoriteCitySection {...props} />
      </Router>
  ).toJSON();
  expect(wrapper).toMatchSnapshot();
});
