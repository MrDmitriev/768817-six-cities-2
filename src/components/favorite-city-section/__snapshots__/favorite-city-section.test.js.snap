// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapShot 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="#"
      >
        <span>
          AAA
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <article
      className="favorites__card place-card"
    >
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          onMouseEnter={[Function]}
          src="aaa"
          width="260"
        />
      </div>
      <div
        className="__card-info place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button place-card__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "NaN%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/undefined"
            onClick={[Function]}
          />
        </h2>
        <p
          className="place-card__type"
        >
          AAA
        </p>
      </div>
    </article>
  </div>
</li>
`;
